pages/admin/lessons/new/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import _ from 'lodash'
+import { AdminLessonCard } from '../../../../components/admin/lessons/AdminLessonCard'
+import {
+  withGetApp,
+  GetAppProps,
+  useGetFlaggedExercisesQuery,
+  GetFlaggedExercisesQuery
+} from '../../../../graphql'
+import { AdminLayout } from '../../../../components/admin/AdminLayout'
+import Link from 'next/link'
+import styles from '../../../../scss/adminLessonPage.module.scss'
+
+type Lesson = {
```

Comment:
dont we already have a Lesson type? It should be in the graphql/index file

Reply:
The `Lesson` type from `graphql `doesn't match the lessons object type from the `getApp` query so it will throw a type-error . It needs to match the type shown here: https://github.com/garageScript/c0d3-app/blob/a5ad3d3b10112324ae9cc35c080c57706c1ba2a4/graphql/index.tsx#L697 

,but without the array. So my current solution was to create a Lessons object mirroring what's shown in the link.

Reply:
Hmm looks the same to me. The Lesson type in the index file has currentUsers and users as fields, but those are optional fields so that shouldn't throw any errors
https://github.com/garageScript/c0d3-app/blob/a5ad3d3b10112324ae9cc35c080c57706c1ba2a4/graphql/index.tsx#L90

So when you use this Lesson type instead of the one you created, you say there is a type error?

Reply:
And when you use the Lesson type from index file , you use it as `Lesson[]` right?

Reply:
![image](https://user-images.githubusercontent.com/77421872/178387483-682707c3-6e77-40d8-b856-c20f9da0402a.png)
i use `Lesson[]` like shown above when when using the `Lesson` type directly imported from `graphql/index`. 

These are the main errors showing :
 `Type '{ __typename?: "Challenge" | undefined; id: number; title: string; description: string; order: number; }[]' is not assignable to type 'Challenge[]'.`
   `Property 'lessonId' is missing in type '{ __typename?: "Challenge" | undefined; id: number; title: string; description: string; order: number; }' but required in type 'Challenge'.`

It appears the `Challenge` field from the imported `Lesson` type is not compatible with the `Lesson` type defined in the `GetAppQuery`
![image](https://user-images.githubusercontent.com/77421872/178388336-ef062828-89fc-4d1a-8360-99c1e38fcfc6.png)

Reply:
I don't have much time to think of further proposals, but here are two proposals without needing to rewrite the entire Lesson type in this file:

1) Do what @flacial suggested and add lessonId to getAppQuery
2) Try to use typescript utils to delete fields you don't want

Example:
```
interface Todo {
  title: string;
  description: string;
  completed: boolean;
  createdAt: number;
  cat: Cat
}

interface Cat {
  nose: string
  removeMePls: string
}

type removedCat = Omit<Cat, "removeMePls">;


type TodoInfo = Omit<Todo, "cat"> & { cat: removedCat }

const todoInfo: TodoInfo = {
  title: "Pick up kids",
  description: "Kindergarten closes at 5pm",
  completed: false,
  createdAt: 1615544252770,
  cat: {
    nose: 'lol'
  },
};
```

I'm not super familiar with typescript, so i'm not sure this is how to best use typescript for this situation. There are maybe other ways more appropriate to implement for this situation than my given proposal.

Reply:
@anthonykhoa I ended up going with this stackoverflow: https://stackoverflow.com/questions/47914536/use-partial-in-nested-property-with-typescript

Basically modify the `Lessons` type from graphql to be able to accept partial keys in the `AdminLessonCard` file.
![image](https://user-images.githubusercontent.com/77421872/179883074-e47fd523-82c9-4c0c-afc3-a5d804185633.png)